src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -161,6 +161,13 @@ public int SendTimeout
             }
         }
 
+        internal bool IsDisconnected { get; private set; } = false;
+
+        internal void SetToDisconnected()
+        {
+            IsDisconnected = true;
+        }
```

Comment:
This doesn't have the same lifetime and accessibility. Socket._isDisconnected can be set to true or false from SetToConnected or SetToDisconnected, while SafeCloseSocket.IsDisconnected can only be set to true from SetToDisconnected. This is because on Unix, once you have disconnected a socket, you can't reuse it (and it gets replaced anyway)